cmake/Platform/Mac/Install_mac.cmake

```diff
@@ -70,31 +64,74 @@ function(ly_install_target_override)
+        
+        ly_file_read(${LY_ROOT_FOLDER}/cmake/Platform/Mac/runtime_install_mac.cmake.in template_file)
+        string(CONFIGURE "${template_file}" configured_template_file @ONLY)
+        file(GENERATE
+            OUTPUT ${CMAKE_BINARY_DIR}/runtime_install/$<CONFIG>/${ly_platform_install_target_TARGET}.cmake
+            CONTENT "${configured_template_file}"
+        )
     endif()
 endfunction()
 
 #! ly_install_add_install_path_setreg: Adds the install path setreg file as a dependency
-function(ly_install_add_install_path_setreg NAME)
-    set_property(TARGET ${NAME} APPEND PROPERTY INTERFACE_LY_TARGET_FILES "${installed_binaries_setreg_path}\nRegistry")
+function(ly_install_add_install_path_setreg NAME SETREG_FILE_PATH)
+    set_property(TARGET ${NAME} APPEND PROPERTY INTERFACE_LY_TARGET_FILES "${SETREG_FILE_PATH}\nRegistry")
 endfunction()
 
 #! ly_install_code_function_override: Mac specific copy function to handle frameworks
 function(ly_install_code_function_override)
 
-    install(CODE
-"function(ly_copy source_file target_directory)
-    if(\"\${source_file}\" MATCHES \"\\\\.[Ff]ramework[^\\\\.]\")
+    configure_file(${LY_ROOT_FOLDER}/cmake/Platform/Mac/InstallUtils_mac.cmake.in ${CMAKE_BINARY_DIR}/runtime_install/InstallUtils_mac.cmake @ONLY)
+    install(SCRIPT ${CMAKE_BINARY_DIR}/runtime_install/InstallUtils_mac.cmake COMPONENT ${CMAKE_INSTALL_DEFAULT_COMPONENT_NAME})
 
-        # fixup origin to copy the whole Framework folder
-        string(REGEX REPLACE \"(.*\\\\.[Ff]ramework).*\" \"\\\\1\" source_file \"\${source_file}\")
-        get_filename_component(target_filename \"\${source_file}\" NAME)
+endfunction()
 
-    endif()
-    file(COPY \"\${source_file}\" DESTINATION \"\${target_directory}\" FILE_PERMISSIONS ${LY_COPY_PERMISSIONS})
-endfunction()"
-    COMPONENT ${CMAKE_INSTALL_DEFAULT_COMPONENT_NAME}
+#! ly_post_install_steps: Any additional platform specific post install steps
+function(ly_post_install_steps)
+
+    # On Mac, after CMake is done installing, the code signatures on all our built binaries will be invalid.
+    # We need to now codesign each dynamic library, executable, and app bundle. It's specific to each target
+    # because there could potentially be different entitlements for different targets.
+    get_property(all_targets GLOBAL PROPERTY LY_ALL_TARGETS)
+    foreach(alias_target IN LISTS all_targets)
+        ly_de_alias_target(${alias_target} target)
+        # Exclude targets that dont produce runtime outputs
+        get_target_property(target_type ${target} TYPE)
+        if(NOT target_type IN_LIST LY_TARGET_TYPES_WITH_RUNTIME_OUTPUTS)
+            continue()
+        endif()
+        
+        install(SCRIPT ${CMAKE_BINARY_DIR}/runtime_install/$<CONFIG>/${target}.cmake COMPONENT ${CMAKE_INSTALL_DEFAULT_COMPONENT_NAME})
+    endforeach()
+
+    install(CODE
+"ly_download_and_codesign_sdk_python()
+ly_codesign_sdk()"
+        COMPONENT ${CMAKE_INSTALL_DEFAULT_COMPONENT_NAME}
     )
```

Comment:
yeah, this is not needed anymore, instead, look at `ly_install_files` and `ly_install_run_code`

Reply:
Will take a look.

Reply:
Looks like ly_install_run_code() only supports CODE type. I can change the install(CODE) at the bottom to use ly_install_run_code(). The install(SCRIPT) is meant to be run at install time. It is similar to CODE but instead of taking the code as a string, it takes a path to a script file. In the above case, for each (runtime)target, we are running the script that we generated in ly_install_target_override(). These scripts do the codesigning for each target. This needs to be run at the end of the cmake install process because cmake uses the install_name_tool to make changes during install.

Reply:
if you want `install(SCRIPT`, lets add a function like `ly_install_run_code`
The reason for these functions is to have a common place for things outside of the install scripts to deal with installing stuff
In this particular case, looks like we are putting O3DE SDK app specific stuff into the install scripts, with a `ly_install_run_script` we could put that logit in the O3DE_SDK CMakeLists.txt file

Reply:
Ok. I'll add `ly_install_run_script` to handle this.